rules/kubecf/create_sample_values.rb

```diff
@@ -0,0 +1,635 @@
+#!/usr/bin/env ruby
+
+# This script creates a sample values.yaml (with most values commented out) from
+# a given values.yaml.
+
+# Usage:
+# create_sample_values.rb input-values.yaml [output-values.yaml]
+
+# The input values.yaml can have comments before mapping keys; they are detected
+# based on indenting.  Any inputs with a comment including the word "REQUIRED"
+# (in all upper case) will be retained as uncommented.  Any inputs with a
+# comment including the words "HIDDEN" (in all upper case) will be
+# omitted from the output.
+
+# If the output file is not given, the result is printed on standard out.
+
+# If the input file is not given, tests are run instead.  If the environment
+# variable `DEBUG` is set, and the input file is given, internal debugging
+# information is printed (to standard output) instead of the adjusted result.
+
+require 'yaml'
+
+class Position
+    def initialize(line, column)
+        @line = line
+        @column = column
+    end
+    attr_reader :line, :column
+    def <=>(other)
+        [line, column] <=> [other.line, other.column]
+    end
+    def <(other)
+        (self <=> other) < 0
+    end
+    def >(other)
+        (self <=> other) > 0
+    end
+    def ==(other)
+        (self <=> other) == 0
+    end
+    def to_s
+        "#{line},#{column}"
+    end
+    def -@
+        return Position.new(-line, -column)
+    end
+end
+
+class Psych::Nodes::Node
+    # We want a bunch of fields on Node to hold state with.
+    attr_accessor :previous_node # Head comment lives between this node & previous
+    attr_accessor :next_node # Line comment lives between this node & next
+    attr_accessor :depth # top level is 1
+    attr_accessor :parent_node, :required
+    attr_writer :start_pos, :end_pos
+    attr_accessor :line_map # for debugging
+    attr_accessor :head_comment, :line_comment
+
+    def start_pos
+        @start_pos ||= Position.new(start_line, start_column)
+    end
+
+    def end_pos
+        @end_pos ||= Position.new(end_line, end_column)
+    end
+
+    def has_children?
+        !(children.nil? || children.empty?)
+    end
+
+    def comment
+        [head_comment, line_comment].compact.reject(&:empty?).join("\n")
+    end
+
+    # These two functions are used to assert preconditions:
+    # state(sym) flags a node has having had an operation done.
```

Comment:
-`has having had`
+`as having had` 
?

Reply:
Fixed in f1e098498638e42df8e9e2999d17f98289a728c3.